singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java

```diff
@@ -221,19 +222,85 @@ private void includeAuditMessageInBatch(List<ProducerRecord<byte[], byte[]>> mes
     return buckets;
   }
 
+  public List<List<LogMessage>> logMessagesCollation(List<PartitionInfo> partitions,
+                                                     String topic,
+                                                     List<LogMessage> logMessages) throws Exception {
+    LOG.info("Collate " + logMessages.size() + " messages");
```

Comment:
Would this be too much logging?

Reply:
Default log level is can control the volume of logging